onnxruntime/core/providers/cpu/math/gemm.h

```diff
@@ -42,6 +42,9 @@ class Gemm final : public OpKernel {
     int64_t N = helper.N();
     int64_t K = helper.K();
     auto Y = context->Output(0, TensorShape({M, N}));
+    // if input is emtpy tensor, return directly as nothing need to be calculated.
```

Comment:
put this right after M and M initialization (line 43).

typo: emtpy -> empty.

Reply:
I think you can't skip line 44, as it will create the MLValue for the output, although the output is an empty tensor...